srtcore/api.cpp

```diff
@@ -263,10 +361,15 @@ SRTSOCKET CUDTUnited::newSocket()
       throw CUDTException(MJ_SYSTEMRES, MN_MEMORY, 0);
    }
 
-   enterCS(m_IDLock);
-   ns->m_SocketID = -- m_SocketIDGenerator;
-   leaveCS(m_IDLock);
-
+   try
+   {
+      ns->m_SocketID = generateSocketID();
+   }
+   catch (...)
+   {
+       delete ns;
+       throw;
+   }
```

Comment:
What is the reason for catching any exception?
`generateSocketID()` only throws `CUDTException`
What are other possible exceptions?

Reply:
This is a simple passthrough - whatever exception (altho' yes, currently it's only `CUDTException`) is thrown, it only gets intercepted here, but then propagated further.

This instruction simply intercepts everything and complements the logical requirements of this function. It doesn't take part in any exception handling at all. If you want to put any real name here, then it would be needed to be treated as this action is specific for the exception - whereas it's not. Whatever gets through it, the handling is completely general.

It's not the matter of being strict, it's the matter of level of details that this very function focuses on.